### PR TITLE
Modified code to get number active PDNs and removed unwanted parameters

### PR DESCRIPTION
**Pull request by rsarwad**
[lte] Modified code to get number active PDNs and removed unwanted parameters

## Summary
There were 3 parameters to represent number of active pdns. 
These were part of,
1.emm_context.esm_ctx. n_pdns
2. emm_context.esm_ctx.n_active_pdns and
3. ue_context_p->nb_active_pdn_contexts
First parameter was used
Second parameter was not used anywhere
Third parameter would be the right place, as UE contexts contains array of PDN contexts so it makes sense to keep number of active PDNs within UE context and not part of esm context.

Modified the code by retaining third parameter and removed unwanted parameters and modified the code accordingly
## Test Plan
Executed s1ap tester sanity testsuite

### CODE REVIEW

**Comment by ssanadhya** [comment]
RestartIntegrationTests

**Comment by ssanadhya** [comment]
RestartLTETests

File: lte/gateway/c/oai/tasks/nas/esm/PdnDisconnect.c [hunk 1]
```diff
@@ -95,7 +95,9 @@ int esm_proc_pdn_disconnect_request(
   mme_ue_s1ap_id_t ue_id =
       PARENT_STRUCT(emm_context, struct ue_mm_context_s, emm_context)
           ->mme_ue_s1ap_id;
-
+  int nb_active_pdn_contexts =
```
Comment (rsarwad) [inline lte/gateway/c/oai/tasks/nas/esm/PdnDisconnect.c]: Get UE context once and get required parameters

File: lte/gateway/c/oai/tasks/nas/esm/esmData.h [hunk 2]
```diff
@@ -167,7 +167,6 @@ typedef struct esm_data_context_s {
   mme_ue_s1ap_id_t ue_id;
 
   int n_ebrs;        /* Total number of active EPS bearer contexts   */
```
Comment (rsarwad) [inline lte/gateway/c/oai/tasks/nas/esm/esmData.h]: Remove the file from git repo

File: lte/protos/oai/nas_state.proto [hunk 3]
```diff
@@ -356,10 +356,8 @@ message EsmProcData {
 // esm_context_t
 message EsmContext {
   uint32 n_active_ebrs = 1;
-  uint32 n_active_pdns = 2;
-  uint32 n_pdns = 3;
-  bool is_emergency = 4;
-  EsmProcData esm_proc_data = 5;
+  bool is_emergency = 2;
```
Comment (rsarwad) [inline lte/protos/oai/nas_state.proto]: Field 2 and 3 are deprecated

**Comment by ssanadhya** [comment]
RestartIntegrationTests

**Comment by ssanadhya** [comment]
RestartLTETests

**Review by ssanadhya — APPROVED** [review]

**Pull request merged**